pythonPackages.pytest-order: init at 0.9.4

jacg opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Package not yet available in nixpkgs

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

jacg committed:
pythonPackages.pytest-order: init at 0.9.3

dotlambda commented on pkgs/development/python-modules/pytest-order/default.nix:
```diff
@@ -0,0 +1,24 @@
+  propagatedBuildInputs = [ pytest ];
+
+  meta = {
+    description = "pytest plugin that allows you to customize the order in which your tests are run";
```
```suggestion
    description = "Pytest plugin that allows you to customize the order in which your tests are run";
```
https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

dotlambda reviewed (changes requested):
"Ran 0 tests in 0.000s"

dotlambda commented:
Will this be used in some other package you're about to create a pull request for? Then it would be nicer to do a single pull request for all of them.

dotlambda reviewed (changes requested):
Commit message and pull request title should start with `pythonPackages`.

jacg commented:
> Will this be used in some other package you're about to create a pull request for? 

No. It's used in the tests of some projects I've come across, but those are unlikely to be submitted to `nixpkgs`.

dotlambda commented:
> "Ran 0 tests in 0.000s"

You need to run the tests. If they are not distributed on PyPI, you can use `fetchFromGitHub`.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110141` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.pytest-order</li>
<li>python38Packages.pytest-order</li>
<li>python39Packages.pytest-order</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.pytest-order:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.pytest-order:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.pytest-order:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>

jacg committed:
WIP: Run tests by fetching from GitHub rather than PyPI

Two tests failing, struggling to understand why.

dotlambda commented on pkgs/development/python-modules/pytest-order/default.nix:
```diff
@@ -0,0 +1,37 @@
+    pytest
+    pytest-xdist
+    pytest-dependency
+    pytest-mock
```
Only `pytest` should be in `propagatedBuildInputs`: https://github.com/mrbean-bremen/pytest-order/blob/v0.9.3/setup.py#L30
The rest should be put into `checkInputs`.

jacg committed:
Move pytest plugins from propagatedBuildInputs to checkInputs

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110141` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.pytest-order: log <a href="https://termbin.com/j4cd">https://termbin.com/j4cd</a></li>
<li>python38Packages.pytest-order: log <a href="https://termbin.com/8ro2">https://termbin.com/8ro2</a></li>
<li>python39Packages.pytest-order: log <a href="https://termbin.com/8yoy">https://termbin.com/8yoy</a></li>
  </ul>
</details>

jacg commented:
There is apparently one pending requested change here, but I cannot find any that requested changes that are unresolved.

jacg committed:
0.9.3 -> 0.9.4 after tests included in new release

After discussion with the repo owner, the tests were included in the PyPI
release. This seems to have fixed all problems with execution of the tests in
the fetch-from-github approach.

jacg commented:
After brief discussion with the maintainer of `pytest-order`, a new release (0.9.4) has been made which includes the tests. I have reverted back to using `fetchPypi` and the tests now pass.

dotlambda commented on pkgs/development/python-modules/pytest-order/default.nix:
```diff
@@ -0,0 +1,37 @@
+    sha256 = "0b7i8z6rywnkb3skyg8bnfqgkjrwvkn64b4q07wfl1q7x65ksd26";
+  };
+
+  checkPhase = false;
```
```suggestion
```

jacg committed:
Don't skip checkPhase

Co-authored-by: Robert Schütz <rschuetz17@gmail.com>
